ddtrace/internal/opentelemetry/metrics.py

```diff
@@ -1,52 +1,215 @@
 import os
+from typing import Any
+from typing import Optional
+from typing import Type
 
+import opentelemetry.version
+
+from ddtrace import config
+from ddtrace.internal.hostname import get_hostname
 from ddtrace.internal.logger import get_logger
+from ddtrace.internal.telemetry import telemetry_writer
+from ddtrace.internal.telemetry.constants import TELEMETRY_NAMESPACE
+from ddtrace.settings._opentelemetry import otel_config
 
 
 log = get_logger(__name__)
 
 
+MINIMUM_SUPPORTED_VERSION = (1, 15, 0)
+API_VERSION = tuple(int(x) for x in opentelemetry.version.__version__.split(".")[:3])
+
+DD_METRICS_PROVIDER_CONFIGURED = False
+
+
 def set_otel_meter_provider():
-    """
-    Get the selected OTLP exporter
-    This function is intended to be called during application startup.
-    """
-    protocol = os.environ.get(
-        "OTEL_EXPORTER_OTLP_PROTOCOL", os.environ.get("OTEL_EXPORTER_OTLP_METRICS_PROTOCOL", "grpc").lower()
-    )
+    """Set up the OpenTelemetry Metrics exporter if not already configured."""
+    if not _should_configure_metrics_exporter():
+        return
+
+    resource = _build_resource()
+    if resource is None:
+        return
+
+    protocol = otel_config.exporter.METRICS_PROTOCOL
+    exporter_class = _import_exporter(protocol)
+    if exporter_class is None:
+        return
+
+    if not _initialize_metrics(exporter_class, protocol, resource):
+        return
+
+    # telemetry_writer.add_count_metric(TELEMETRY_NAMESPACE.TRACERS, "metrics_provider_configured", 1, (("type", "dd"),))
```

Comment:
I am planning to remove this metric from the python tracer. We can just rely on the configurations sent to metabase.

https://github.com/DataDog/dd-trace-py/pull/14721